File: BST/python3/tree.py
from typing import Any

from node import Node


class Tree:
    def __init__(self) -> None:
        self.root: Node | None = None

    def add(self, value) -> bool:
        if self.root is None:
            self.root = Node(value)
            return True
        return self.root.add(value)

    def contain(self, value) -> bool:
        if self.root is None:
            return False
        return self.root.contain(value)

    def length(self) -> int:
        if self.root is None:
            return 0
        return self.root.length()

    def height(self) -> int:
        if self.root is None:
            return 0
        return self.root.height()

File: BST/cs/Tree.cs
class Tree
{
    public Node root;

    public Tree(){}

    public bool add(int value){
        if(this.root == null){
            this.root = new Node(value);
            return true;
        }
        return this.root.add(value);
    }

    public bool contain(int value){
        if(this.root == null){
            return false;
        }
        return this.root.contain(value);
    }

    public int length(){
        if(this.root == null){
            return 0;
        }
        return this.root.length();
    }

    public int height(){
        if(this.root == null){
            return 0;
        }
        return this.root.height();
    }
}
